### PR TITLE
DP-23183 Style rich text horizontal separator on info details page

### PR DESCRIPTION
**Pull request by Javi-er**
Any PRs being created needs a changelog.txt file before being merged into dev. See: [Change Log Instructions](https://github.com/massgov/mayflower/blob/develop/docs/for-developers/changelog-instructions.md)

## Description

Adds divider as a component, adds variant for thin, converts it to hr.
## Related Issue / Ticket

- [JIRA issue](https://massgov.atlassian.net/browse/DP-23183)

## Steps to Test

1. Checkout both MF and OpenMass branches.
2. Visit a few organizations and info detail pages and validate that 'hr' is the tag used for the line
3. Validate that the margins and paddings are correct.

## Screenshots
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.

## Additional Notes:

Anything else to add?

#### Impacted Areas in Application

*

#### @TODO

*

### CODE REVIEW

File: packages/patternlab/styleguide/source/_patterns/01-atoms/divider.md [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
-
-### Status
-* Stable as of 5.0.0
-
-### Variables
-~~~
-none
-~~~
+### Description
+This is a simple horizontal divider.  It is typically used at the bottom of the Page Header.
```
Comment (clairesunstudio) [inline packages/patternlab/styleguide/source/_patterns/01-atoms/divider.md]:
```suggestion
This is a simple horizontal divider.  It is typically used at the bottom of a section.
```

File: packages/patternlab/styleguide/source/_patterns/01-atoms/divider.md [hunk 2]
```diff
@@ -1,10 +1,13 @@
...
+
+### Status
+* Stable as of 5.0.0
```
Comment (clairesunstudio) [inline packages/patternlab/styleguide/source/_patterns/01-atoms/divider.md]:
```suggestion
* Stable as of 11.18.0
```

File: changelogs/DP-23183.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+#    description: Describe the change and add the PR number. see ___DESCRIPTION___ e.g. Adds apples to apple trees for admin apple pickers (#PR)
+#    issue: Add a Jira ticket or issue number, e.g. DP-12345 or 124
+#    impact: Enter a valid impact, e.g. Minor
+
+#___EXAMPLE___
+Fixed:
+  - project: Patternlab
+    component: Divider
+    description: Adds divider as a component, adds variant for thin, converts it to hr. (#1541)
+    issue: DP-17253
```
Comment (clairesunstudio) [inline changelogs/DP-23183.yml]:
```suggestion
    issue: DP-23183
```

File: packages/patternlab/styleguide/source/_patterns/01-atoms/divider.twig [hunk 4]
```diff
@@ -1 +1 @@
-<div class="ma__divider"></div>
+<hr class="ma__divider {{ (divider.thin == true)?"ma__divider--thin":'' }}" />
```
Comment (clairesunstudio) [inline packages/patternlab/styleguide/source/_patterns/01-atoms/divider.twig:1]: Do you know why the markup is showing `<hr></hr>`
Screen Shot 2021-11-02 at 1 33 15 PM

This is actually invalid according to https://validator.w3.org/nu/#textarea
I feel like the thin divider shouldn't need this the :after sudo class either.

**Comment by clairesunstudio** [comment]
#### The alignment on info details page looks a lot better, but there's a little margin left/right of 4px applied on the :after class, which should be removed and taken care of after eliminating :after for the thing version of divider.

Screen Shot 2021-11-02 at 4 55 38 PM
#### Also in this example, do we need to increase the margin top/bottom on the divider?
Screen Shot 2021-11-02 at 4 36 37 PM


#### On org pages, the divider is not aligned with the content. Can you see if the margins/paddings can be added to align with the content?

Screen Shot 2021-11-02 at 5 00 29 PM
Screen Shot 2021-11-02 at 5 00 37 PM




----

### Links to compare:
##### Org page: 
- https://www.mass.gov/orgs/office-of-the-inspector-general
- https://pr1079-fn1xgidofccolywkusp1xftcnggfttdn.tugboat.qa/orgs/office-of-the-inspector-general
##### Info details page: 
- https://www.mass.gov/info-details/massachusetts-clean-energy-and-climate-plan-for-2025-and-2030
- https://pr1079-fn1xgidofccolywkusp1xftcnggfttdn.tugboat.qa/info-details/massachusetts-clean-energy-and-climate-plan-for-2025-and-2030
- https://www.mass.gov/info-details/qag-information-details1
- https://pr1079-fn1xgidofccolywkusp1xftcnggfttdn.tugboat.qa/info-details/qag-information-details1

**Comment by clairesunstudio** [comment]
Please look at the visual regressions, I think some of the divider under pageHeader is cut off in the backstop tests
https://43551-137521197-gh.circle-artifacts.com/0/backstop/reports/html/index.html

Screen Shot 2021-11-02 at 5 10 49 PM

Update: going through those look like they are on org and org detail pages, can you update those org pages and templates in mayflower to use the thin divider instead? 

After that, we might want to push the openmass branch to a feature environment and run backstop it

File: packages/patternlab/styleguide/source/_patterns/01-atoms/divider.twig (hunk 4 above)
Comment (Javi-er) [reply packages/patternlab/styleguide/source/_patterns/01-atoms/divider.twig:1]: j-r-capture 2021-11-04 a la(s) 09 33 14
That's something that the inspector is doing, if you check the actual source code, there isn't a separate closing tag.

Comment (Javi-er) [reply packages/patternlab/styleguide/source/_patterns/01-atoms/divider.twig:1]: regarding the :after, it's needed for the padding on the container to have effect.

**Comment by Javi-er** [comment]
the 4px padding and the visual regressions where fixed already.

File: packages/patternlab/styleguide/source/_patterns/01-atoms/divider.twig (hunk 4 above)
Comment (clairesunstudio) [reply packages/patternlab/styleguide/source/_patterns/01-atoms/divider.twig:1]: interesting that the inspect is showing that

**Comment by clairesunstudio** [comment]
New report https://43576-137521197-gh.circle-artifacts.com/0/backstop/reports/html/index.html

The diffs look fine to me!
Except for this one, looks like the change has increased the vertical spacing, can you fix this. And approve the screenshots once this one is fixed. 
Screen Shot 2021-11-04 at 1 48 57 PM

File: packages/assets/scss/01-atoms/_divider.scss [hunk 5]
```diff
@@ -50,3 +44,11 @@
     }
   }
 }
+
+.ma__org-page .ma__divider,
+.ma__organization-detail .ma__divider{
+  @include ma-container;
+
+//  padding: 0;
```
Comment (clairesunstudio) [inline packages/assets/scss/01-atoms/_divider.scss]:
```suggestion
//  padding: 0;
```
looks good, can you just clean this line up

**Comment by Javi-er** [comment]
I just fixed that padding issue and also removed the commented out line too.

**Review by clairesunstudio — APPROVED** [review]

**Comment by clairesunstudio** [comment]
false postive test

**Pull request merged**